presto-product-tests/src/main/java/com/facebook/presto/tests/utils/JdbcDriverUtils.java

```diff
@@ -78,7 +78,11 @@ else if (usingTeradataJdbcDriver(connection)) {
 
     public static boolean usingPrestoJdbcDriver(Connection connection)
     {
-        return getClassNameForJdbcDriver(connection).equals("com.facebook.presto.jdbc.PrestoConnection");
+        String canonicalClassName = getClassNameForJdbcDriver(connection);
```

Comment:
@arhimondr, @maciejgrzybek: This affects the usingTeradataJdbc*Driver methods too because they also get the canonical classname of the connection. Do we want to switch those to isinstance or look up the canonical class name of the super class if we get null back.

Bearing in mind that we don't control the name of the TD JDBC Connection object, I'm inclined to go with looking up the super class's canonical name. It's hacky, but (probably) less brittle.